### PR TITLE
Project api

### PR DESCRIPTION
**Pull request by DevDelvo**
- [x] get all projects for one user (public)
- [ ]create project by a user route (private, and validated route)
    - [x] take in an array of images -> s3 (s3-multer (back-end package))
    - [x] upload an image - multi-part form request (form data use conventions - images[])
    - [x] take in the other basic details of project - project title, etc.
- [x] S3 - set AWS credentials -> do not ever post your AWS credentials on github (gitignore .env file from now on) - share credentials over slack

### CODE REVIEW

File: server/routes/index.js
Comment (tsyaeger) [inline server/routes/index.js]: Related to your auth routes comment -- will the plan be to just have a routes file for each model and add the protected route middleware to each route?

Comment (DevDelvo) [reply server/routes/index.js]: yeah!

File: server/controllers/Project.js
Comment (evankp) [inline server/controllers/Project.js]: Probably should remove this if not being used

File: server/controllers/authController.api.js
Comment (evankp) [inline server/controllers/authController.api.js]: Not sure if this comment is really needed to be honest

File: server/routes/Project.js
Comment (evankp) [inline server/routes/Project.js]: Unless you are still working on it, maybe remove this.

File: server/routes/index.js
Comment (evankp) [inline server/routes/index.js]: I see you put this into a separate file, maybe remove it here?

**Review by evankp — CHANGES_REQUESTED** [review]
Let me know how soon you want to merge this if you are finished or not, Conflicts with my #25

File: server/routes/Project.js
Comment (tsyaeger) [inline server/routes/Project.js]: can we just combine the verifyJwt and isAuth?

File: server/controllers/Project.js
Comment (DevDelvo) [reply server/controllers/Project.js]: I left it because it was the initial way I had wanted to do it but then Shums mentioned the populate method. However, I'm not sure if this is actually what he wanted (although both give us all the projects associated with the user). I'm going to clean it up after stand up for another review before merging

File: server/controllers/authController.api.js
Comment (DevDelvo) [reply server/controllers/authController.api.js]: Sorry that was a note! lol

File: server/routes/Project.js
Comment (DevDelvo) [reply server/routes/Project.js]: I think we can do that! I'll refactor it once I'm home later today

File: server/routes/Project.js
Comment (DevDelvo) [reply server/routes/Project.js]: Yeah, I will remove it

File: server/controllers/Project.js
Comment (skassam21) [inline server/controllers/Project.js]: Yes this is what i'm looking for - however I wanted this in the get user route (I commented about it below). The idea was we don't really need this route, but we need a route to get the users and the projects together

File: server/controllers/authController.api.js
Comment (skassam21) [inline server/controllers/authController.api.js]: I know this is not your code, but we shouldn't be returning 500s in the code

File: server/controllers/authController.api.js
Comment (skassam21) [inline server/controllers/authController.api.js]: Are using cookies or localstorage?

File: server/controllers/authController.api.js
Comment (skassam21) [inline server/controllers/authController.api.js]: let's get the user and projects together

File: server/controllers/authController.api.js
Comment (evankp) [reply server/controllers/authController.api.js]: Probably should be 400

**Comment by evankp** [comment]
@DevDelvo Let me know once you want me to do another review and I'll go through and look it over to approve

File: server/controllers/authController.api.js
Comment (DevDelvo) [reply server/controllers/authController.api.js]: I'll change this to a 400 error then

File: server/controllers/authController.api.js
Comment (DevDelvo) [reply server/controllers/authController.api.js]: I'll remove this since we've agreed to just use localstorage for now

File: server/controllers/authController.api.js
Comment (DevDelvo) [reply server/controllers/authController.api.js]: okay I can move the populate to this method then

**Comment by evankp** [comment]
@DevDelvo I think you need to merge dev into branch, unless I am missing where you did

File: server/controllers/Project.js
Comment (evankp) [inline server/controllers/Project.js:21]: Nothing blocking this from being merged, but we should discuss how we send error json, IE as array or single object.

File: server/controllers/authController.api.js
Comment (evankp) [inline server/controllers/authController.api.js]: If we have res.status I do not think we need status in the json response data, as we can read response code on front end

File: server/controllers/authController.api.js
Comment (evankp) [inline server/controllers/authController.api.js]: Maybe move status to before .json?
```
return res.status(200).json({
                    token,
                    user: {
                        _id,
                        name,
                        email,
                        projects
                    }
                });
```

File: server/models/User.js
Comment (evankp) [inline server/models/User.js]: I changed name to not be unique, as I thought we discussed we can have similar names between users, but email unique

File: server/controllers/authController.api.js
Comment (evankp) [inline server/controllers/authController.api.js]: I think Shums mentioned that we should not use 500 error codes actually, but I do know this was carried over from when the route was created

Comment (DevDelvo) [reply server/controllers/authController.api.js]: Ah, I didn't see that one! What code should we use in this case?

Comment (evankp) [reply server/controllers/authController.api.js]: I think 400, but a 4xx error I guess? 500 is Internal Server Error

**Review by evankp — APPROVED** [review]
Most of the comments I added now are not blocking it being merged now, more just stying and error code choices

File: server/controllers/Project.js
Comment (skassam21) [reply server/controllers/Project.js:21]: 👍 in the past I've created helper functions to send error messaging

File: server/controllers/Project.js
Comment (skassam21) [inline server/controllers/Project.js:49]: This could probably also throw an error

File: server/controllers/authController.api.js
Comment (skassam21) [reply server/controllers/authController.api.js]: Probably take a look at why it would fail and it'll give us a better sense of the error code - it may be because the user doesn't exist (in case 404), or because of a bad formatted name (400), etc. 

Sometimes people throw 500s, but it's really a catch all and should be avoided

File: server/controllers/authController.api.js
Comment (skassam21) [inline server/controllers/authController.api.js:45]: this is more of a 400

File: server/controllers/authController.api.js
Comment (skassam21) [reply server/controllers/authController.api.js]: You didn't populate in this method yet

**Review by skassam21 — APPROVED** [review]
Looks good! Nothing really blocking, so feel free to address these and then merge. We will improve as we go

**Pull request merged**
